src/nursery/client_cfg.ml

```diff
@@ -84,4 +84,17 @@ module ClientCfg = struct
     in
     cfg
 
+  let _from_file section filename =
+    Lwt_extra.read_file filename >>= fun txt ->
+    _from_txt section txt |> Lwt.return
+
+  let _from_etcd section peers path =
+    Arakoon_etcd.retrieve_value peers path >>= fun txt ->
+    _from_txt section txt |> Lwt.return
+
+  let from_url section url =  (* This is the format as defined in the extension *)
+    match url with
```

Comment:
This pattern match should really be done only once. (Now I see it three times in the pull request)